bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -125,28 +125,28 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 		return ctrl.Result{}, err
 	}
 
-	// Look up the Machine that owns this KubeConfig if there is one
-	machine, err := util.GetOwnerMachine(ctx, r.Client, config.ObjectMeta)
+	// Look up the owner of this KubeConfig if there is one
+	configOwner, err := util.GetConfigOwner(ctx, r.Client, config.ObjectMeta)
 	if err != nil {
-		log.Error(err, "could not get owner machine")
+		log.Error(err, "could not get owner")
 		return ctrl.Result{}, err
 	}
-	if machine == nil {
-		log.Info("Waiting for Machine Controller to set OwnerRef on the KubeadmConfig")
+	if configOwner == nil {
+		log.Info("Waiting for the configuration owner's controller to set OwnerRef on the KubeadmConfig")
 		return ctrl.Result{}, nil
 	}
-	log = log.WithValues("machine-name", machine.Name)
+	log = log.WithValues("config-owner-name", configOwner.GetName())
```

Comment:
Would it make sense to pull the type information from the unstructured resource and add it to the logging in addition to the name?
maybe something like
```go
log = log.WithValues("kind", configOwner.Kind, "version", configOwner.GetResourceVersion(), "name", configOwner.GetName())?
```